src/cli/tc/check.rs

```diff
@@ -8,28 +10,45 @@ use crate::core::config::reader::ConfigReader;
 use crate::core::config::Source;
 use crate::core::runtime::TargetRuntime;
 
-pub(super) struct CheckParams {
-    pub(super) file_paths: Vec<String>,
-    pub(super) n_plus_one_queries: bool,
-    pub(super) schema: bool,
-    pub(super) format: Option<Source>,
-    pub(super) runtime: TargetRuntime,
+pub struct CheckParams {
+    pub file_paths: Vec<String>,
+    pub n_plus_one_queries: bool,
+    pub schema: bool,
+    pub format: Option<Source>,
+    pub runtime: TargetRuntime,
 }
 
-pub(super) async fn check_command(params: CheckParams, config_reader: &ConfigReader) -> Result<()> {
+pub async fn check_command(
+    params: CheckParams,
+    config_reader: &ConfigReader,
+    mut write_buf: Option<&mut dyn Write>,
```

Comment:
I think we can hook in a custom tracing module in the tests. We can then assert what was written to the buffer.